### PR TITLE
Feature/display activities pictures

### PR DESCRIPTION
**Pull request by GhaliBerbich**
This PR restructures the interaction with Firestore Storage for image management in the app to align with the MVVM architecture pattern. The following changes were made:

Created ImageRepository Interface:

Abstracts the image storage functionality for easier mocking and testing.
Added ImageRepositoryFirestore Implementation:

Implements ImageRepository to interact with Firestore Storage, handling image upload, retrieval, and deletion.
Introduced ImageViewModel:

Manages the UI-related logic for images and bridges the gap between the repository and the UI.
Modified Existing Screens:

Updated the signatures of affected screens to include the new ImageViewModel.
Ensured compatibility with the refactored image handling logic.
Enabled Mocking for Tests:

The new structure allows mocking of ImageRepository in tests, resolving previous CI failures related to image fetching.
As a result, all existing tests now pass.
Tests:

Refactored tests for impacted features to use the new structure.
Previous CI failures due to unmocked Firestore interactions have been resolved.
Pending Tasks:
Add unit tests specifically for the ImageViewModel to ensure comprehensive test coverage.
Impact:
This refactor improves code modularity, makes testing easier, and ensures compliance with the MVVM architecture. It fixes previous CI issues related to image fetching and prepares the codebase for future enhancements in image management.

### CODE REVIEW

**Comment by GhaliBerbich** [comment]
You can accept my PR now, I will add tests later tonight, but for the moment to avoid merge conflicts (that I had to resolve 3 times for multiple files, since I made my PR), merge my work please if it is conveniant to the standards we have.

File: app/src/main/java/com/android/sample/model/image/CameraActions.kt [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    e.printStackTrace()
+    null
+  }
+}
+
+fun Bitmap.resize(reqWidth: Int): Bitmap {
+  val ratio: Float = this.width.toFloat() / this.height.toFloat()
+  val height = (reqWidth / ratio).toInt()
+  return Bitmap.createScaledBitmap(this, reqWidth, height, true)
+}
```
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/model/image/CameraActions.kt:65]: I tried on my emulator and everything work, very well

File: app/src/main/java/com/android/sample/model/image/ImageRepository.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      onSuccess: (List<String>) -> Unit,
+      onFailure: (Exception) -> Unit
+  )
+
+  fun fetchActivityImagesAsBitmaps(
+      activityId: String,
+      onSuccess: (List<Bitmap>) -> Unit,
+      onFailure: (Exception) -> Unit
+  )
+}
```
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/model/image/ImageRepository.kt:37]: Good implementation, you follow the MVVM very precisely !

File: app/src/main/java/com/android/sample/model/image/ImageRepositoryFirestore.kt [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+                  bitmaps.add(bitmap)
+                  successCount++
+                  if (successCount == urls.size) onSuccess(bitmaps)
+                }
+                .addOnFailureListener { onFailure(it) }
+          }
+        },
+        onFailure)
+  }
+}
```
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/model/image/ImageRepositoryFirestore.kt:145]: You can replace hardcoded strings with constants for better scalability and error prevention.
The fact that you have 25 and 100 is a bit confusing

File: app/src/main/java/com/android/sample/model/image/ImageViewModel.kt
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/model/image/ImageViewModel.kt:1]: LGTM

File: app/src/main/java/com/android/sample/ui/activity/EditActivity.kt [hunk 4]
```diff
@@ -122,16 +123,20 @@ fun EditActivityScreen(
...
-  var selectedImages = remember { mutableStateListOf<Bitmap>() }
-
+  var selectedImages by remember { mutableStateOf<List<Bitmap>>(emptyList()) }
+  var items by remember { mutableStateOf(activity?.images ?: listOf()) }
+  imageViewModel.fetchActivityImagesAsBitmaps(
+      activity?.uid ?: "",
+      { bitmaps -> selectedImages = bitmaps.toMutableStateList() },
+      onFailure = { error ->
+        Log.e("EditActivityScreen", "Failed to fetch images: ${error.message}")
+      })
```
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/ui/activity/EditActivity.kt:134]: LGTM

File: app/src/main/java/com/android/sample/ui/camera/CameraPreview.kt [hunk 5]
```diff
@@ -88,3 +88,19 @@ fun Carousel(openDialog: () -> Unit, itemsList: List<Bitmap>, deleteImage: (Bitm
...
+        Card(modifier = Modifier.padding(4.dp)) {
+          Image(
+              bitmap = itemsList[bitmap].resize(100).asImageBitmap(),
+              contentDescription = "Selected Image",
+              modifier = Modifier.size(100.dp))
+        }
+      }
+    }
+  }
+}
```
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/ui/camera/CameraPreview.kt:106]: It is a good implementation, it has a very good UI.
But please try to remove the magic numbers

File: app/src/main/java/com/android/sample/ui/profile/EditProfile.kt [hunk 6]
```diff
@@ -205,12 +210,37 @@ fun EditProfileScreen(profileViewModel: ProfileViewModel, navigationActions: Nav
...
+                            User(
+                                id = profile.id,
+                                name = name,
+                                surname = surname,
+                                interests = newListInterests,
+                                activities = profile.activities,
+                                photo = photo,
+                                likedActivities = profile.likedActivities))
+                        navigationActions.goBack()
+                      } catch (_: NumberFormatException) {}
```
Comment (aminebenji04) [inline app/src/main/java/com/android/sample/ui/profile/EditProfile.kt:237]: LGTM !

**Review by aminebenji04 — CHANGES_REQUESTED** [review]
In general everything is good, you mastered MVVM, good job! However if you see my comments you will notice that I requested you to remove magic constants and try to make your choice of the values 25 and 100 more explicit in ImageRepositoryFirestore

**Review by aminebenji04 — APPROVED** [review]

**Pull request merged**